### DISCUSSION TITLE
Unable to reset password

### DISCUSSION THREAD

**Discussion by isaacgr** [firefly-iii/firefly-iii#6887] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Similar issue to previous reports about not being able to reset their password. 

I did the steps listed in #1611. This worked to get me a reset link, however when I submit my new password I get 

> Error! There were some problems with your input.
> This password reset token is invalid.

On the page

There doesn't appear to be any output in the logs as to why this is.

### Expected behaviour

The password reset link should work.

### Debug information

I cant access this.

**Comment by JC5** [comment]
I'm not sure if instructions from 2018 are still valid though 😉 

I can't seem to replicate this, sorry. You're saying the email arrives just fine but then the link doesn't work?

**Reply by TRuHa83** [reply]
Hello, I have exactly the same problem.

I have tried to use the method that you mention in #1611 checking the logs where it returns an address with a token but it always returns the same error:

**This password reset token is invalid.**

Any ideas?

thanks for everything.

**Comment by isaacgr** [comment]
Not sure if its like its a date setting thing? My server and PC are in EST. The firefly app is in CET and the mysql db is in UTC. Not sure what the expiry time is on the token. Doing the conversion to EST the app and db are the same time. The time that gets inserted into the db is the CET time of the firefly app which makes sense.

**Reply by isaacgr** [reply]
> Not sure if its like its a date setting thing? My server and PC are in EST. The firefly app is in CET and the mysql db is in UTC. Not sure what the expiry time is on the token. Doing the conversion to EST the app and db are the same time. The time that gets inserted into the db is the CET time of the firefly app which makes sense.

I tried changing the timezone of my computer to match the app to test this but no luck.

**Comment by isaacgr** [comment]
> I'm not sure if instructions from 2018 are still valid though 😉
> 
> I can't seem to replicate this, sorry. You're saying the email arrives just fine but then the link doesn't work?

Yes thats correct. The link allows me to get to the reset page, but then says its invalid when i submit a new password.

![image](https://user-images.githubusercontent.com/20640588/212890222-ea5f9cde-83b6-4fe6-aeac-896eb16a70e9.png)

**Comment by c4baceira** [comment]
I am having the same issue. Checked the email in the log file and I can get the link but with the expired token. Currently locked out of the admin account.

**Comment by JC5** [comment]
I can't replicate this, and there aren't much logs. Firefly III will always be vague about whether or not it sent a password reset request, you'll have to check the log files if anything was sent. The URL you're getting will have a 64 character token. All of its validation is handled by Laravel, I'm not yet sure what the root could be.

**Comment by kristoferbi** [comment]
I am having the same problem. I have copied the link from the log because I don't have the mail set up properly yet and I am getting the exact same error.

EDIT: I noticed that there were 65 characters in my token, including **=**. I removed the **=** and then I was prompted for the 2FA code, which of course I also don't have, but that's a separate problem.

**Reply by c4baceira** [reply]
Check in the logs if the token is breaking into the next line or if you can spot some CRLF in between. I think it was some concatenation issue.

I remember the fix was something like that, the token not being properly displayed in the logs and ending up wrong when copy/paste. 

My bad for not returning here when it was fresh.
